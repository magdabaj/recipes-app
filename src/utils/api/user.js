const api = 'https://afternoon-plateau-23579.herokuapp.com/user';
// const api = 'http://localhost:5000/user'

export const signUpUser = async (email, password) => {
    return fetch(`${api}/signup`, {
        method: 'POST',
        body: JSON.stringify({ email, password }),
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};

export const signInUser = async (email, password) => {
    return fetch(`${api}/signin`, {
        method: 'POST',
        body: JSON.stringify({ email, password }),
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};

export const signOutUser = async () => {
    return fetch(`${api}/signout`, {
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};

export const authenticateUser = async () => {
    return fetch(`${api}/authenticated`, {
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};
