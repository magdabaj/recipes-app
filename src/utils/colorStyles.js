const colorStyles = {
    darkRed: '#5D001E',
    lightGrey: '#E3E2DF',
    lightPink: '#E3AFBC',
    mediumPink: '#EE4C7C',
    darkPink: '#9A1750',
};

export default colorStyles;
