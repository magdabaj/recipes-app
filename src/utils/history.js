// import { createBrowserHistory } from 'history';
//
// const history = createBrowserHistory();

import { createMemoryHistory } from "history";

const history = createMemoryHistory()

export default history;
