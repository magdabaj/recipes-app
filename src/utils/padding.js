const basePadding = 8;

const padding = {
    extraSmall: `${basePadding / 4}px`, // 2px
    small: `${basePadding / 2}px`, // 4px
    base: `${basePadding}px`, // 8px
    medium: `${basePadding * 1.5}px`, // 12px
    large: `${basePadding * 2}px`, // 16px
}

export default padding;