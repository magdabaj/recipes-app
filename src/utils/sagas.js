import { all } from 'redux-saga/effects';
import { watchRecipesByTagLoad, watchRecipesLoad, watchTagsLoad } from "../containers/HomePage/saga";

export default function* rootSaga() {
    yield all([
        watchRecipesLoad(),
        watchTagsLoad(),
        watchRecipesByTagLoad()
    ]);
}