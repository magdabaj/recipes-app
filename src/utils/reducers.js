import { combineReducers } from 'redux';
import { connectRouter } from "connected-react-router";
import homePageReducer from "../containers/HomePage/reducer";
import loginContainerReducer from "../containers/LoginContainer/reducer";
import userRecipesContainerReducer from "../containers/UserRecipesContainer/reducer";
import recipesFormContainerReducer from "../containers/RecipesFormContainer/reducer";
import recipeContainerReducer from "../containers/RecipeContainer/reducer";
import appReducer from '../containers/App/reducer';
import history from "./history";

const rootReducer = combineReducers({
    homePageReducer,
    loginContainerReducer,
    userRecipesContainerReducer,
    recipesFormContainerReducer,
    recipeContainerReducer,
    appReducer,
    router: connectRouter(history),
})

export default rootReducer;