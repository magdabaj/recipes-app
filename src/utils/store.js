import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import reducers from './reducers';
import sagas from './sagas';

const sagaMiddleware = createSagaMiddleware();
// const middleware = [sagaMiddleware];

const composeEnhancers =
    typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
            // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
        })
        : compose;

const store = createStore(
    reducers,
    composeEnhancers(applyMiddleware(sagaMiddleware)),
    // +  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
sagaMiddleware.run(sagas);
export default store;