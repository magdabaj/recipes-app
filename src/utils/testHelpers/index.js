import { Router } from 'react-router-dom'
import { createMemoryHistory } from "history";
import store from "../store";
import {Provider} from "react-redux";
import React from "react";
import { render } from "@testing-library/react";
import {createStore} from "redux";
// import {reducer} from "react-toastify/dist/hooks/toastContainerReducer";

// const store  = createStore(rootReducer, initialState={})
const memoryHistory = createMemoryHistory({
    initialEntries: ['/'],
})

export const withStoreAndRouter = component =>
    <Router history={memoryHistory}>
        <Provider store={store}>
            {component}
        </Provider>
    </Router>

export const renderWithRouter = (
    ui,
    {
        history = memoryHistory,
    } = {}
) => (
    {
        ...render(<Router history={history}>{ui}</Router>),
        history,
    }
)

export const renderWithRouterKentCDodds = (
    ui,
        {
            route='/',
            history = createMemoryHistory({initialEntries: [route]}),
            ...renderOptions
        } = {}
) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Router history={history}>
            {children}
        </Router>
    }

    return {
        ...render(ui, {wrapper: Wrapper, ...renderOptions})
    }
}

export const renderWithProviderKentCDodds = (
        ui,
        reducer,
        {initialState, store = createStore(reducer, initialState), ...options} = {},
    ) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
    return render(ui, {wrapper: Wrapper, ...options})
}