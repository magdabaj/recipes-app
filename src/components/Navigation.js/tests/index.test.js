import React from 'react';
import { render } from "@testing-library/react";
import { withStoreAndRouter } from "../../../utils/testHelpers";
import Navigation from "../index";
import {fakeUser} from "../../../utils/testHelpers/fixtures/user";

describe('<Navigation/>', () => {
    const renderNavigation = (signOut = jest.fn(), user = fakeUser) =>
        render(
            withStoreAndRouter(
                <Navigation signOut={signOut} user={user}/>
            )
        )

    it('should not log errors in console', function () {
        const spy = jest.spyOn(global.console, 'error')
        renderNavigation()
        expect(spy).not.toHaveBeenCalled()
    });

    it('should render and match the snapshot', function () {
        const {
            container: { firstChild }
        } = renderNavigation()
        expect(firstChild).toMatchSnapshot()
    });
})