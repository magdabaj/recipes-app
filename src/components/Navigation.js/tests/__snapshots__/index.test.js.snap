// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> should render and match the snapshot 1`] = `
<div
  class="navigation-container"
>
  <div
    class="navigation"
  >
    <div
      class="navigation-title"
    >
      <a
        aria-current="page"
        class="navigation-title--item active"
        href="/"
      >
        Przepisy
      </a>
    </div>
    <div
      class="navigation-links"
    >
      <a
        aria-current="page"
        class="navigation-links--item navigation-links--active"
        href="/"
      >
        Strona główna
      </a>
      <a
        class="navigation-links--item"
        href="/user-recipes"
      >
        Twój profil
      </a>
    </div>
    <div
      class="navigation-login"
    >
      <div
        class="navigation-login--item"
      >
        <div
          class="logout"
        >
          Wyloguj
        </div>
      </div>
    </div>
  </div>
</div>
`;
